shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm

```diff
@@ -82,8 +80,81 @@ - (void)swapBuffers {
             _frameBuffers[kFlutterSurfaceManagerFrontBuffer]);
 }
 
-- (uint32_t)glFrameBufferId {
-  return [_frameBuffers[kFlutterSurfaceManagerBackBuffer] glFrameBufferId];
+- (FlutterBackingStoreDescriptor*)renderBufferDescriptor {
+  uint32_t fboId = [_frameBuffers[kFlutterSurfaceManagerBackBuffer] glFrameBufferId];
+  return [[FlutterBackingStoreDescriptor alloc] initOpenGLDescriptorWithFBOId:fboId];
+}
+
+@end
+
+@implementation FlutterMetalSurfaceManager {
+  CGSize _surfaceSize;
+  id<MTLDevice> _mtlDevice;
+  id<MTLCommandQueue> _mtlCommandQueue;
+  CAMetalLayer* _containingLayer;  // provided (parent layer)
+  CALayer* _contentLayer;
+
+  FlutterIOSurfaceHolder* _ioSurfaces[kFlutterSurfaceManagerBufferCount];
```

Comment:
There's non-trivial duplication between the two surface managers; perhaps superclass to manage the common parts would be useful.